### ISSUE TITLE
Using BeamSearchDecoder for Language Modeling (Smart Compose)

### ISSUE THREAD

**Issue by StarWang** [tensorflow/addons#2346]
**Describe the feature and the current behavior/state.**
Hi I wonder if the BeamSearchDecoder provided in tfa.seq2seq can be used for language modeling. If so, how do we implement it?

My current thought is to use the same cell from encoder and decoder. But it seems that passing the rnn cell from encoder to decoder during decoder initialization is not the right way to do it. If this is the way to solve it, can you share some scripts on how to do that? Thanks!

**Relevant information**
- Are you willing to contribute it (yes/no): no
- Are you willing to maintain it going forward? (yes/no): no
- Is there a relevant academic paper? (if so, where): https://www.kdd.org/kdd2019/accepted-papers/view/gmail-smart-compose-real-time-assisted-writing
- Is there already an implementation in another framework? (if so, where): no
- Was it part of tf.contrib? (if so, where): no

**Which API type would this fall under (layer, metric, optimizer, etc.)**
layer

**Who will benefit with this feature?**
Users interested in language modeling

**Any other info.**

**Comment by chazuttu** [comment]
# beam search
def beam_search_decoder(data, k):
	sequences = [[list(), 0.0]]
	# walk over each step in sequence
	for row in data:
		all_candidates = list()
		# expand each current candidate
		for i in range(len(sequences)):
			seq, score = sequences[i]
			for j in range(len(row)):
				candidate = [seq + [j], score - log(row[j])]
				all_candidates.append(candidate)
		# order all candidates by score
		ordered = sorted(all_candidates, key=lambda tup:tup[1])
		# select k best
		sequences = ordered[:k]
	return sequences

**Comment by chazuttu** [comment]
beam_search_decoder() function performs the beam search decoder

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
